xdg-terminal-exec

```diff
@@ -303,229 +299,176 @@ alias find_entry_paths='IFS= find_entry_paths'
 	key="$1"
 	value="$2"
-	action="$3"
-	read_exec="$4"
-	de_checks="$5"
+	action_id="$3"
 
-	# Order of checks is important
 	case $key in
-	'Categories'*=*)
-		debug "checking for 'TerminalEmulator' in Categories '$value'"
-		IFS=';'
-		for category in $value; do
-			[ "$category" = "TerminalEmulator" ] && {
-				TERMINAL=true
-				return 0
-			}
-		done
-		# Default in this case is to fail
-		return 1
-		;;
-	'Actions'*=*)
-		# `It is not valid to have an action group for an action identifier not mentioned in the Actions key.
-		# Such an action group must be ignored by implementors.`
-		# ignore if no action requested
-		[ -z "$action" ] && return 0
-		debug "checking for '$action' in Actions '$value'"
-		IFS=';'
-		for check_action in $value; do
-			[ "$check_action" = "$action" ] && return 0
-		done
-		# Default in this case is to fail
-		return 1
-		;;
-	'OnlyShowIn'*=*)
-		case "$de_checks" in
-		true) debug "checking for intersecion between '${XDG_CURRENT_DESKTOP-}' and OnlyShowIn '$value'" ;;
-		false)
-			debug "skipping OnlyShowIn check"
-			return 0
-			;;
-		esac
-		IFS=';'
-		for target in $value; do
-			IFS=':'
-			for desktop in ${XDG_CURRENT_DESKTOP-}; do
-				[ "$desktop" = "$target" ] && return 0
-			done
-		done
-		# Default in this case is to fail
-		return 1
-		;;
-	'NotShowIn'*=*)
-		case "$de_checks" in
-		true) debug "checking for intersecion between '${XDG_CURRENT_DESKTOP-}' and NotShowIn '$value'" ;;
-		false)
-			debug "skipping NotShowIn check"
-			return 0
-			;;
-		esac
-		IFS=';'
-		for target in $value; do
-			IFS=':'
-			for desktop in ${XDG_CURRENT_DESKTOP-}; do
-				debug "checking NotShowIn match '$desktop'='$target'"
-				[ "$desktop" = "$target" ] && return 1
-			done
-		done
-		# Default in this case is to succeed
-		return 0
-		;;
-	'X-ExecArg'*=* | 'ExecArg'*=*)
-		# Set global variable
-		EXECARG=$value
-		debug "read ExecArg '$EXECARG'"
-		;;
-	'TryExec'*=*)
-		debug "checking TryExec executable '$value'"
-		command -v "$value" > /dev/null || return 1
-		;;
-	'Hidden'*=*)
-		debug "checking boolean Hidden '$value'"
-		[ "$value" = 'true' ] && return 1
-		;;
-	'Exec'*=*)
-		case "$read_exec" in
-		false)
-			debug "ignored Exec from wrong section"
-			return 0
-			;;
-		esac
-		debug "read Exec '$value'"
-		# Set global variable
-		EXEC=$value
-		# Get first word from read Exec value
-		IFS="$OIFS"
-		eval "set -- $EXEC"
-		debug "checking Exec[0] executable '$1'"
-		command -v "$1" > /dev/null || return 1
-		;;
+	'X-ExecArg' | 'ExecArg') check_entry_execarg "$value" ;;
+	'Categories') check_entry_categories "$value" ;;
+	'OnlyShowIn') check_entry_onlyshowin "$value" ;;
+	'NotShowIn') check_entry_notshowin "$value" ;;
+	'Actions') check_entry_actions "$value" "$action_id" ;;
+	'TryExec') check_entry_tryexec "$value" ;;
+	'Hidden') check_entry_hidden "$value" ;;
+	# Only read key from requested action group, marked by unset ID
+	'Exec') [ -z "$action_id" ] && check_entry_exec "$value" ;;
```

Comment:
This fails on any explicitly specified action, needs to be wrapped in full `if` `then` `fi`